kwok/tools/gen_instance_types.go

```diff
@@ -0,0 +1,121 @@
+/*
```

Comment:
I'm figuring you're thinking about this, but it would be awesome if we had follow-ups to this PR that auto-gened test data from AWS and Azure cloudproviders as well so you could do a bunch of test simulating with data from those providers!

Reply:
I have one for AWS, I've done very little with Azure so don't have a generator for them yet.  Agree it would be nice!